app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Adapters/FavoritesListAdapter.java

```diff
@@ -0,0 +1,147 @@
+package ithaca_transit.android.cornellappdev.com.ithaca_transit.Adapters;
+
+import android.content.Context;
+import android.os.health.SystemHealthManager;
+import android.support.v7.widget.RecyclerView.Adapter;
+import android.support.v7.widget.RecyclerView.ViewHolder;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.View.OnClickListener;
+import android.widget.TextView;
+
+import com.appdev.futurenovajava.APIResponse;
+import com.appdev.futurenovajava.Endpoint;
+import com.appdev.futurenovajava.FutureNovaRequest;
+
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.Models.BusStop;
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.Models.Favorite;
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.Models.Route;
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.R;
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.Singleton.Repository;
+import ithaca_transit.android.cornellappdev.com.ithaca_transit.Utils.FutureUtilities;
+import kotlin.TypeCastException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+
+public final class FavoritesListAdapter extends Adapter {
+
+    // stores all routes for favorite at the position represented by the key
+    private HashMap<Integer, ArrayList<Route>> mAllRoutesToFavorites = new HashMap<Integer, ArrayList<Route>>();
+
+    private Endpoint.Config mConfig;
+    private Context mContext;
+    private Favorite[] mFavList;
+    private final FavoritesListAdapter.ListAdapterOnClickHandler mListAdapterOnClickHandler;
+
+    // stores most optimal route for favorite
+    private Route[] mOptimalRoutes;
+    private static Repository REPOSITORY;
+
+    public FavoritesListAdapter(@NotNull Context context, @NotNull ListAdapterOnClickHandler listAdapterOnClickHandler,
+                                @NotNull Favorite[] favorites, Endpoint.Config config, Repository repository) {
+        super();
+        mConfig = config;
+        mContext = context;
+        mFavList = favorites;
+        mListAdapterOnClickHandler = listAdapterOnClickHandler;
+        mOptimalRoutes = new Route[favorites.length];
+        REPOSITORY = repository;
```

Comment:
You don't need to add the `Repository` as a param because it can be called directly in this class. 

Also the way we deal with the repo in Eatery is creating `Repository` instance inside itself and creating a `getInstance()` function: 

```private static final Repository sRepoInstance = new Repository();
  private boolean isSearchPressed = false;

  public static Repository getInstance() {
    return sRepoInstance;
  }
```
And then inside this class, you can call: 
```
private Repository rInstance = Repository.getInstance();
```